libraries/AP_Baro/AP_Baro_SITL.cpp

```diff
@@ -20,6 +20,7 @@ AP_Baro_SITL::AP_Baro_SITL(AP_Baro &baro) :
 #if APM_BUILD_TYPE(APM_BUILD_ArduSub)
         _frontend.set_type(_instance, AP_Baro::BARO_TYPE_WATER);
 #endif
+        set_bus_id(_instance, AP_HAL::Device::make_bus_id(AP_HAL::Device::BUS_TYPE_SITL, 0, _instance, DEVTYPE_BARO_SITL));
```

Comment:
No `set_device_type`?

Reply:
> `set_device_type`

we don't have a dev to call it on, make_bus_id() does all fields